src/pg.cr

```diff
@@ -1,13 +1,18 @@
+require "db"
 require "./pg/*"
 
 module PG
   # Establish a connection to the database
-  def self.connect(conninfo)
-    Connection.new(conninfo)
+  def self.connect(url)
+    DB.open(url)
   end
 
   # Establish a special listen connection to the database
-  def self.connect_listen(conninfo, *channels : String, &blk : PQ::Notification ->)
-    ListenConnection.new(conninfo, *channels, &blk)
+  def self.connect_listen(url, *channels : String, &blk : PQ::Notification ->)
+    db = DB.open(url)
+    db.using_connection do |conn|
+      conn.as(PG::Connection).listen(*channels, &blk)
+    end
+    db
```

Comment:
The reason before that I returned a different class here was to make sure you can't accidentally try and use it to send normal queries on it, because when the driver is in the listen loop it wont work.